src/Illuminate/Foundation/ProviderRepository.php

```diff
@@ -74,7 +74,7 @@ public function load(array $providers)
             $this->app->register($this->createProvider($provider));
         }
 
-        $this->app->setDeferredServices($manifest['deferred']);
+        $this->app->addDeferredServices($manifest['deferred']);
```

Comment:
When loading extra service providers using a ProviderRepository later in the Application, currently it would replace all deferred services that were already registered, rather than just adding additional deferred services.